inject.js

```diff
@@ -551,6 +578,14 @@ function initializeNow(document) {
   });
 }
 
+function setSpeed(controller, video, speed) {
+  var speedvalue = speed.toFixed(2);
+  video.playbackRate = Number(speedvalue);
+  var speedIndicator = controller.shadowRoot.querySelector("span");
+  speedIndicator.textContent = speedvalue;
```

Comment:
Do we still need these two lines given updated logic?

Reply:
Thanks for pointing this out...

Yes, actually this is even more essential now as the ratechange event is blocked, however, this made me realize that now the logic inside of our ratechange event would never be triggered by VSC which I think is the -exact opposite- of the desired behavior.

I have moved this into its own function and now it's triggered irrespective of if the speed change request comes from VSC or if it comes from the site itself. This does bring in to question whether we should be storing lastspeed when it's the website that requested the change rather than VSC, but I've preserved the existing behavior.

If this needs to be changed to only store speeds originating from requests of VSC I certainly don't mind making that change but think it may be better to split into its own bug report since this is how it behaves today.